DESIGN.md

```diff
@@ -337,7 +337,8 @@ type MeshCataloger interface {
 	// GetServicesForServiceAccount returns a list of services corresponding to a service account
 	GetServicesForServiceAccount(service.K8sServiceAccount) ([]service.MeshService, error)
 
-	// GetResolvableHostnamesForUpstreamService returns the hostnames over which an upstream service is accessible from a downstream service
+  // GetResolvableHostnamesForUpstreamService returns the hostnames over which an upstream service is accessible from a downstream service
+  // TODO : remove as a part of routes refactor (#2397)
 	GetResolvableHostnamesForUpstreamService(downstream, upstream service.MeshService) ([]string, error)
```

Comment:
The PR description mentions `GetResolvableHostnamesForUpstreamService` being removed but this is merely being commented on for future removal. Could you clarify?

Reply:
it will eventually be removed once we cut over as a part of #2397 as the current code has dependency on it.

Reply:
I have updated the description to reduce confusion

Reply:
If we get rid of this method name, how will we calculate hostnames?

Reply:
The method `getServiceHostnames` is indirectly being called by `GetResolvableHostnamesForUpstreamService` and if you look at all the code in routes v2, it all calls `getServiceHostnames` . Therefore making it possible for us to remove `GetResolvableHostnamesForUpstreamService` as a part of the cleanup during cut over